app/src/main/java/org/mifos/mobile/ui/update_password/UpdatePasswordViewModel.kt

```diff
@@ -0,0 +1,121 @@
+package org.mifos.mobile.ui.update_password
+
+import android.content.Context
+import android.widget.Toast
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import dagger.hilt.android.lifecycle.HiltViewModel
+import dagger.hilt.android.qualifiers.ApplicationContext
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.launch
+import org.mifos.mobile.R
+import org.mifos.mobile.repositories.ClientRepository
+import org.mifos.mobile.repositories.UserAuthRepository
+import org.mifos.mobile.utils.Network
+import org.mifos.mobile.utils.RegistrationUiState
+import javax.inject.Inject
+
+@HiltViewModel
+class UpdatePasswordViewModel @Inject constructor(
+    private val userAuthRepositoryImp: UserAuthRepository,
+    private val clientRepositoryImp: ClientRepository,
+    @ApplicationContext private val context: Context
```

Comment:
dont pass context in a viewmodel. This may lead to memory leaks

Reply:
_regular contexts can leak to memory leaks because their life cycle might be shorter than that of the viewmodel, but applicationContext is an exception because it's alive as long as the application is_

Could you please recheck it?

Reply:
https://developer.android.com/reference/androidx/lifecycle/AndroidViewModel

Reply:
Application context is just less likely to lead to memory leaks , it isn't entirely immune to the leaks. Moreover passing context makes it difficult to test a viewmodel

Reply:
ohh.. ok bro.. Then how can we access string resources in ViewModel then? Any idea?

Reply:
That's the point, we don't access anything android framework dependent in VM. You can access them in your composable if needed or use the context in your fragment/activity

Reply:
ohh! I will make the changes

Reply:
@PratyushSingh07 I have made the changes.

Reply:
[Screencast_04_16_2024_08:21:57_AM.webm](https://github.com/openMF/mifos-mobile/assets/78101731/5ff86bd0-83a0-4c5c-8efb-6cb1eed40b89)
also I think we have to use duration... snackbar not disappearing